### PR TITLE
Feat: RedeemManager Migration - Assumes  initiator equals recipient

### PR DESCRIPTION
**Pull request by koderholic**
## Description
The current redeemManager migration for v1.2.1 upgrade requires an array of external provided addresses to be set for the initiator field of exiting redeemRequests without initiator. This could potentially introduce a margin for error, due to getting off-chain data into the protocol.

This PR creates a method of migrating which DOES NOT require any external data to be passed into the migration. 

## Notice

- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/liquid-collective/liquid-collective-protocol/pulls) for the same update/change?
- [x] Have you assigned this PR to yourself?
- [x] Have you added at least 1 reviewer?
- [ ] Have you updated the official documentation?
- [x] Have you added sufficient documentation in your code?
- [x] Have you added relevant tests to the official test suite?

## Pull Request Type

- [ ] 💫 New Feature (Breaking Change)
- [x] 💫 New Feature (Non-breaking Change)
- [ ] 🛠️ Bug fix (Non-breaking Change: Fixes an issue)
- [ ] 🕹️ Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)

## Testing

- [ ] Have you tested this code with the official test suite?
- [x] Have you tested this code manually?

### CODE REVIEW

**Review by iamsahu — CHANGES_REQUESTED** [review]
Looks good, just a minor comment addition & getting the code coverage up.

File: contracts/test/RedeemManager.1.t.sol [hunk 1]
```diff
@@ -2087,17 +2081,17 @@ contract InitializeRedeemManagerV1_2Test is RedeeManagerV1TestBase {
...
 
     function testRedeemQueueMigrationV1_2() public {
         // Call the migration function
         RedeemManagerV1 redeemQueueImplV2 = new RedeemManagerV1();
         vm.store(redeemManager, IMPLEMENTATION_SLOT, bytes32(uint256(uint160(address(redeemQueueImplV2)))));
-        RedeemManagerV1(redeemManager).initializeRedeemManagerV1_2(prevInitiators);
+        RedeemManagerV1(redeemManager).initializeRedeemManagerV1_2();
 
         // Check all existing redeemRequests are intact after the migration  (from oldQueue)
         for (uint256 i = 0; i < 30; i++) {
```
[the remark this answers is not in the document]
Comment (koderholic) [reply contracts/test/RedeemManager.1.t.sol:2097]: Sure! I can add that, there's no significance behind the numbers though, it's a random choice.

**Review by iamsahu — DISMISSED** [review]
🚀

**Review by ATMackay — APPROVED** [review]

**Review by juliaaschmidt — APPROVED** [review]
🚀 🚀

**Review by mischat — APPROVED** [review]
LGTM

**Pull request merged**